src/context.rs

```diff
@@ -34,10 +36,13 @@ impl<'a> Context<'a> {
             .map(|entry| entry.path())
             .collect::<Vec<PathBuf>>();
 
+        let repository: Option<Repository> = Repository::discover(&current_dir).ok();
+
         Context {
-            current_dir,
             arguments,
+            current_dir,
             dir_files,
+            repository,
```

Comment:
I'm kind of just adding things that would reduce redundant operations across segments. 😅 

I know that we'll end up using `Context` to share user configuration across segments, and maybe a list of binaries accessible via `$PATH` (so that we don't have to call `Command::new()` when it doesn't make sense to). I don't foresee much else being added after this point.

Can you think of a better way to organize `Context`?